seahub/api2/endpoints/admin/sys_notifications.py

```diff
@@ -0,0 +1,147 @@
+        """
+
+        try:
+            notifications = Notification.objects.all().order_by('-id')
+        except Exception as e:
+            error_msg = 'Internal Server Error'
+            return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, error_msg)
+
+        results = []
+        for notification in notifications:
+            results.append(get_notification_info(notification))
+
+        return Response({'notifications': results})
+
+    def post(self, request):
+        """
+        create a system notification
+
+        Permission checking:
+        1.login and is admin user.
+        """
+
+        msg = request.data.get('msg', '')
+        if not msg:
+            error_msg = 'msg invalid.'
+            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+        try:
+            notification = Notification.objects.create_sys_notification(msg)
+        except Exception as e:
+            logger.error(e)
+            error_msg = 'Internal Server Error'
+            return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, error_msg)
+
+        notification_info = get_notification_info(notification)
+
+        return Response({'notification': notification_info})
+
+
+class AdminSysNotificationView(APIView):
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    throttle_classes = (UserRateThrottle,)
+    permission_classes = (IsAuthenticated, IsAdminUser)
+
+    def put(self, request, nid):
+        """
+        update a system notification primary status
+
+        Permission checking:
+        1.login and is admin user.
+        """
+
+        try:
+            nid = int(nid)
+        except ValueError:
+            error_msg = 'nid invalid.'
+            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+        if nid <= 0:
+            error_msg = 'nid invalid.'
+            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+        try:
+            Notification.objects.filter(primary=1).update(primary=0)
+            Notification.objects.filter(id=nid).update(primary=1)
+        except Exception as e:
+            logger.error(e)
+            error_msg = 'Internal Server Error'
+            return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, error_msg)
+
+        cache.set('CUR_TOPINFO', Notification.objects.all().filter(primary=1),
+                  NOTIFICATION_CACHE_TIMEOUT)
+
+        return Response({'success': True})
+
+    def delete(self, request, nid):
+        """
+        delete a system notification
+
+        Permission checking:
+        1.login and is admin user.
+        """
+
+        try:
+            nid = int(nid)
+        except ValueError:
+            error_msg = 'nid invalid.'
+            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+        if nid <= 0:
+            error_msg = 'nid invalid.'
+            return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+        try:
+            Notification.objects.get(id=nid)
```

Comment:
noti..._list = Notification.objects.filter(id=nid)

if not n_list:


for n in n_list